### PR TITLE
Add coverage command and PR annotation

### PR DESCRIPTION
**Pull request by nadzyah**
This PR simply modifies the server GH action and adds coverage report to it

### CODE REVIEW

File: .github/workflows/test_server.yaml [hunk 1]
```diff
@@ -44,6 +44,33 @@ jobs:
       - name: Run tests with docker-compose
         run: |
           docker-compose up --attach-dependencies --abort-on-container-exit hwapi-test
+          sudo chown $USER -R coverage
+          sudo chgrp $USER -R coverage
+      - name: Upload coverage.xml as an artifact
+        uses: actions/upload-artifact@v3
```
Comment (omar-selo) [inline .github/workflows/test_server.yaml:50]: any reason not to use v4?

File: .github/workflows/test_server.yaml [hunk 2]
```diff
@@ -44,6 +44,33 @@ jobs:
...
+          if-no-files-found: error
+          retention-days: 2
+
+  report-coverage:
+    runs-on: [self-hosted, linux, large, jammy, x64]
+    if: github.event_name == 'pull_request'
+    needs: [test]
+    steps:
+      - name: Download build tests coverage report
+        uses: actions/download-artifact@v3
```
Comment (omar-selo) [inline .github/workflows/test_server.yaml:63]: same here

**Review by omar-selo — COMMENTED** [review]
Looks good just a couple of comments

File: scripts/generate_openapi_html.sh [hunk 3]
```diff
@@ -5,6 +5,7 @@ if git diff --cached --name-only | grep --quiet "server"
 then
     cd server/schemas/
 
+    which npx
```
[the remark this answers is not in the document]
Comment (nadzyah) [reply scripts/generate_openapi_html.sh]: The problem is that the hook passes even if npx is not installed, that's why I've decided to fix it here and add check that we have this command. I've adjusted the script to exit if it's not found

**Review by omar-selo — APPROVED** [review]
LGTM!

**Pull request merged**